src/imu_control_parameters.cpp

```diff
@@ -0,0 +1,209 @@
+        }
+
+        if(m_data_ready_polarity != data_ready_polarity)
+        {
+            m_iioWrapper.set_data_ready_polarity(data_ready_polarity);
+            m_data_ready_polarity = data_ready_polarity;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "data_ready_polarity: '%d'", data_ready_polarity);
+        }
+
+        if(m_sync_signal_scale != sync_signal_scale)
+        {
+            m_iioWrapper.set_sync_signal_scale(sync_signal_scale);
+            m_sync_signal_scale = sync_signal_scale;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "sync_signal_scale: '%d'", sync_signal_scale);
+        }
+
+        if(m_decimation_filter != decimation_filter)
+        {
+            m_iioWrapper.set_decimation_filter(decimation_filter);
+            m_decimation_filter = decimation_filter;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "decimation_filter: '%d'", decimation_filter);
+        }
+
+        if(m_accel_calibbias_x != accel_calibbias_x)
+        {
+            m_iioWrapper.set_accel_calibbias_x(accel_calibbias_x);
+            m_accel_calibbias_x = accel_calibbias_x;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "accel_calibbias_x: '%d'", accel_calibbias_x);
+        }
+
+        if(m_accel_calibbias_y != accel_calibbias_y)
+        {
+            m_iioWrapper.set_accel_calibbias_y(accel_calibbias_y);
+            m_accel_calibbias_y = accel_calibbias_y;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "accel_calibbias_y: '%d'", accel_calibbias_y);
+        }
+
+        if(m_accel_calibbias_z != accel_calibbias_z)
+        {
+            m_iioWrapper.set_accel_calibbias_z(accel_calibbias_z);
+            m_accel_calibbias_z = accel_calibbias_z;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "accel_calibbias_z: '%d'", accel_calibbias_z);
+        }
+
+        if(m_anglvel_calibbias_x != anglvel_calibbias_x)
+        {
+            m_iioWrapper.set_anglvel_calibbias_x(anglvel_calibbias_x);
+            m_anglvel_calibbias_x = anglvel_calibbias_x;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "anglvel_calibbias_x: '%d'", anglvel_calibbias_x);
+        }
+
+        if(m_anglvel_calibbias_y != anglvel_calibbias_y)
+        {
+            m_iioWrapper.set_anglvel_calibbias_y(anglvel_calibbias_y);
+            m_anglvel_calibbias_y = anglvel_calibbias_y;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "anglvel_calibbias_y: '%d'", anglvel_calibbias_y);
+        }
+
+        if(m_anglvel_calibbias_z != anglvel_calibbias_z)
+        {
+            m_iioWrapper.set_anglvel_calibbias_z(anglvel_calibbias_z);
+            m_anglvel_calibbias_z = anglvel_calibbias_z;
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "anglvel_calibbias_z: '%d'", anglvel_calibbias_z);
+        }
```

Comment:
see my remark in iio wrapper, if setting of parameter does not succeed, an exception should be thrown to let the user know that the setting did not occur, the error should also be made visible to the user, in case the parameter value is invalid

Reply:
It was implemented this logic in another commit amend